IllegalStateException: Unable to serialize SQL type: -155, Object: 2021-07-15 01:00:00 +00:00

`
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.IllegalStateException: Unable to serialize SQL type: -155, Object: 2021-07-15 01:00:00 +00:00
	at pz.tool.jdbcimage.main.MainToolBase.runConcurrently(MainToolBase.java:511)
	at pz.tool.jdbcimage.main.MainToolBase.run(MainToolBase.java:443)
	at pz.tool.jdbcimage.main.MultiTableParallelExport.run(MultiTableParallelExport.java:24)
	at pz.tool.jdbcimage.main.MultiTableParallelExport.main(MultiTableParallelExport.java:52)
	at pz.tool.jdbcimage.main.JdbcImageMain.main(JdbcImageMain.java:18)
Caused by: java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.IllegalStateException: Unable to serialize SQL type: -155, Object: 2021-07-15 01:00:00 +00:00
	at java.base/java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1006)
	at pz.tool.jdbcimage.main.MainToolBase.runConcurrently(MainToolBase.java:509)
	... 4 more
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.IllegalStateException: Unable to serialize SQL type: -155, Object: 2021-07-15 01:00:00 +00:00
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:600)
	... 6 more
Caused by: java.lang.RuntimeException: java.lang.IllegalStateException: Unable to serialize SQL type: -155, Object: 2021-07-15 01:00:00 +00:00
	at pz.tool.jdbcimage.db.QueryRunner.run(QueryRunner.java:50)
	at pz.tool.jdbcimage.main.SingleTableExport.exportTable(SingleTableExport.java:41)
	at pz.tool.jdbcimage.main.MultiTableParallelExport.lambda$getExportTask$1(MultiTableParallelExport.java:33)
	at pz.tool.jdbcimage.main.MainToolBase.lambda$runConcurrently$7(MainToolBase.java:485)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1448)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.IllegalStateException: Unable to serialize SQL type: -155, Object: 2021-07-15 01:00:00 +00:00
	at pz.tool.jdbcimage.kryo.KryoResultSetConsumer.accept(KryoResultSetConsumer.java:132)
	at pz.tool.jdbcimage.kryo.KryoResultSetConsumer.accept(KryoResultSetConsumer.java:19)
	at pz.tool.jdbcimage.db.QueryRunner.run(QueryRunner.java:44)
	... 9 more
`

--- a/src/main/java/pz/tool/jdbcimage/db/DbImportResultConsumer.java
+++ b/src/main/java/pz/tool/jdbcimage/db/DbImportResultConsumer.java
@@ -120,6 +120,7 @@ public class DbImportResultConsumer implements ResultConsumer<RowData>{
                                 break;
                             case Types.CHAR:
                             case Types.VARCHAR:
+                            case Mssql.Types.DATETIMEOFFSET:
                                 stmt.setString(pos, (String)value);
                                 break;
                             case Types.NCHAR:

--- a/src/main/java/pz/tool/jdbcimage/kryo/KryoResultProducer.java
+++ b/src/main/java/pz/tool/jdbcimage/kryo/KryoResultProducer.java
@@ -76,6 +76,7 @@ public class KryoResultProducer implements ResultProducer{
 				case Types.NCHAR:
 				case Types.VARCHAR:
 				case Types.NVARCHAR:
+				case Mssql.Types.DATETIMEOFFSET:
 					val = kryo.readObjectOrNull(in, String.class);
 					break;
 				case Types.DATE:

--- a/src/main/java/pz/tool/jdbcimage/kryo/KryoResultSetConsumer.java
+++ b/src/main/java/pz/tool/jdbcimage/kryo/KryoResultSetConsumer.java
@@ -92,6 +92,7 @@ public class KryoResultSetConsumer implements ResultConsumer<ResultSet> {
                         break;
                     case Types.CHAR:
                     case Types.VARCHAR:
+                    case Mssql.Types.DATETIMEOFFSET:
                         val = rs.getString(i + 1);
                         clazz = String.class;
                         break;

--- a/src/main/java/pz/tool/jdbcimage/main/Mssql.java
+++ b/src/main/java/pz/tool/jdbcimage/main/Mssql.java
@@ -159,8 +159,8 @@ public class Mssql extends DBFacade {
     }
 
     public static class Types {
-
         public static final int SQL_VARIANT = -156;
+        public static final int DATETIMEOFFSET = -155;
 
         private Types() { }
     }

